### PR TITLE
[ENH] Add IndividualLITERegressor and LITETimeRegressor

### LINKED ISSUE #1219: `IndividualLITERegressor`

### LINKED ISSUE #1220: `LITETimeRegressor`

### PR DESCRIPTION
**Pull request by aadya940**
Closes #1219 
Closes #1220

### What does this implement/fix? Explain your changes.

  1.Create _lite_time.py Module in aeon.regression
  2. Call LITENetwork
  3. Replace Ouput-Layer Activation by "linear"
  4. Update Loss to "mse"
  5. Update docstrings

### CODE REVIEW

File: aeon/regression/deep_learning/_lite_time.py [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+
+        self.callbacks = callbacks
+        self.random_state = random_state
+        self.verbose = verbose
+        self.use_mini_batch_size = use_mini_batch_size
+        self.loss = loss
+        self.metrics = metrics
+        self.optimizer = optimizer
+
+        self.classifers_ = []
```
Comment (MatthewMiddlehurst) [inline aeon/regression/deep_learning/_lite_time.py:154]: not classifiers

**Review by MatthewMiddlehurst — COMMENTED** [review]
Just one comment for now, @hadifawaz1999 would know better if these are as expected.

File: aeon/regression/deep_learning/_lite_time.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+
+
+class LITETimeRegressor(BaseRegressor):
+    """LITETime ensemble Regressor.
+
+    Ensemble of IndividualLITETimeRegressor objects, as described in [1]_.
+
+    Parameters
+    ----------
+    n_classifiers : int, default = 5,
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:25]: its n_regressors not n_classifiers

File: aeon/regression/deep_learning/_lite_time.py [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+        save_last_model is set to False, this parameter
+        is discarded
+    random_state : int, default = 0
+        seed to any needed random actions.
+    verbose : boolean, default = False
+        whether to output extra information
+    optimizer : keras optimizer, default = Adam
+    loss : keras loss, default = "mse"
+    metrics : keras metrics, default = None,
+        will be set to accuracy as default if None
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:77]: will be set to "mean_squared_error" instead

File: aeon/regression/deep_learning/_lite_time.py [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+        save_last_model_to_file
+    best_file_name : str, default = "best_model"
+        The name of the file of the best model, if
+        save_best_model is set to False, this parameter
+        is discarded
+    last_file_name : str, default = "last_model"
+        The name of the file of the last model, if
+        save_last_model is set to False, this parameter
+        is discarded
+    random_state : int, default = 0
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:70]: random_state default to None  not 0

File: aeon/regression/deep_learning/_lite_time.py [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+            The training data target values.
+
+        Returns
+        -------
+        self : object
+        """
+        rng = check_random_state(self.random_state)
+
+        for n in range(0, self.n_regressors):
+            cls = IndividualLITERegressor(
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:175]: rgs instead of cls
cls is for classifier and rgs for regressor

File: aeon/regression/deep_learning/_lite_time.py [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+
+        Returns
+        -------
+        Y : np.ndarray of shape = (n_instances (n)), the predicted target values
+
+        """
+        check_random_state(self.random_state)
+        vals = np.zeros(X.shape[0])
+
+        for cls in self.regressors_:
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:215]: rgs

File: aeon/regression/deep_learning/_lite_time.py [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+                use_mini_batch_size=self.use_mini_batch_size,
+                n_epochs=self.n_epochs,
+                callbacks=self.callbacks,
+                loss=self.loss,
+                metrics=self.metrics,
+                optimizer=self.optimizer,
+                random_state=rng.randint(0, np.iinfo(np.int32).max),
+                verbose=self.verbose,
+            )
+            cls.fit(X, y)
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:193]: rgs

File: aeon/regression/deep_learning/_lite_time.py [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+            "n_epochs": 10,
+            "batch_size": 4,
+            "kernel_size": 4,
+        }
+
+        return [param1]
+
+
+class IndividualLITERegressor(BaseDeepRegressor):
+    """Single LITETime Regressor.
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:255]: single LITE regressor, LITETime is the ensemble of LITE

File: aeon/regression/deep_learning/_lite_time.py [hunk 9]
```diff
@@ -0,0 +1,537 @@
...
+        save_last_model_to_file
+    best_file_name      : str, default = "best_model"
+        The name of the file of the best model, if
+        save_best_model is set to False, this parameter
+        is discarded
+    last_file_name      : str, default = "last_model"
+        The name of the file of the last model, if
+        save_last_model is set to False, this parameter
+        is discarded
+    random_state : int, default = 0
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:304]: random_state default to None not 0

File: aeon/regression/deep_learning/_lite_time.py [hunk 10]
```diff
@@ -0,0 +1,537 @@
...
+        seed to any needed random actions.
+    verbose : boolean, default = False
+        whether to output extra information
+    optimizer : keras optimizer, default = Adam
+    loss : keras loss, default = 'mse'
+    metrics : keras metrics, default = None,
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:310]: same comment on the metrics as before

File: aeon/regression/deep_learning/_lite_time.py [hunk 11]
```diff
@@ -0,0 +1,537 @@
...
+        output_layer = tf.keras.layers.Dense(1, activation=self.output_activation)(
+            output_layer
+        )
+
+        model = tf.keras.models.Model(inputs=input_layer, outputs=output_layer)
+
+        tf.random.set_seed(self.random_state)
+
+        if self.metrics is None:
+            metrics = ["accuracy"]
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_lite_time.py:416]: metric should be default to mean_squared_error

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks for this, just some comments to be fixed

**Comment by aadya940** [comment]
Thanks for the Review :-) @MatthewMiddlehurst @hadifawaz1999

**Review by MatthewMiddlehurst — DISMISSED** [review]
Looks like @hadifawaz1999 comments have been resolved. I don't see anything noticeably off, but it's not my code 🙂

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks, it all looks good, last changes are:

1- merge branch main
2- do the changes as we did yesterday in #1292  for the model saving format from hdf5 to keras

**Comment by aadya940** [comment]
@hadifawaz1999 Thanks for the Review :-), The changes have been done.

**Comment by aadya940** [comment]
Done, Thanks :-) @hadifawaz1999

**Comment by aadya940** [comment]
Hey @hadifawaz1999 @MatthewMiddlehurst , The tests are failing in certain python versions but seems to be unrelated to this issue. wdyt?

**Comment by MatthewMiddlehurst** [comment]
@aadya940 apologies for the late reply, yes we were having a general failure on CI. If you update the branch to main this should be resolved.

**Review by hadifawaz1999 — APPROVED** [review]
LGTM !
